src/main/scala/com/microsoft/azure/iot/iothub2kafka/EventHubReceiver.scala

```diff
@@ -12,6 +12,8 @@ import scala.collection.mutable.ListBuffer
 class EventHubReceiver(val connectionString: String, val receiverConsumerGroup: String, val partition: String,
     var offset: Option[String], val startTime: Option[Instant]) extends DataReceiver {
 
+  private[this] var isClosing = false
```

Comment:
could we use `com.microsoft.azure.servicebus.ClientEntity.getIsClosingOrClosed()` ?